### PR TITLE
feat: cell-level keyboard navigation for Table

### PR DESCRIPTION
**Pull request by jacobdevera**
### Description
**Pending update of `fundamental-styles` to incorporate this PR https://github.com/SAP/fundamental-styles/pull/1661 to hide screen reader instructions**
- add Table example that uses interactive content (similar to https://openui5.hana.ondemand.com/entity/sap.ui.table.Table/sample/sap.ui.table.sample.Basic)
- add cell-level keyboard navigation
    - arrow keys to move between cells; if there is one focusable element within the field without any arrow-key handling, it will take focus
    - `Enter` to toggle 'edit mode': in edit mode, all tabbable fields can be navigated to with the Tab key
    - `Esc` to exit edit mode
- add `compact`, `condensed`, `keyboardNavigation`, and `localizedText` props
- use `aria-live` region for announcing table status

### Out of Scope / Known Issues (to be created)
- row-level keyboard navigation
- `Select` should be full-width in cells (likely requires prop passing by putting into a `FormGroup` to apply full-width styling)
- `DatePicker` popover when opened from a cell appears to gradually shift off-screen when changing months; this doesn't happen with the DatePicker alone

### CODE REVIEW

File: src/Table/Table.js [hunk 1]
```diff
@@ -44,17 +61,67 @@ const Table = React.forwardRef(({ headers, tableData, className, tableBodyClassN
...
     }
 
     return (
-        <table {...props} className={tableClasses}
-            ref={ref}>
+        <table {...props} aria-describedBy={captionId}
+            className={tableClasses}
+            ref={tableRef}
+            role={keyboardNavigation ? 'grid' : 'table'}>
+            <caption aria-live='polite' className='fd-table__instructions'
```
Comment (Mike-Diaz) [inline src/Table/Table.js:121]: Is `__instructions` intentional? In fd-styles the class for `<caption>` appears as `fd-table__caption`. Currently the text is visible when traversing through the cells.

File: src/Table/__stories__/Table.stories.js [hunk 2]
```diff
@@ -120,3 +145,103 @@ export const richTable = () => {
...
+                tableRowData.map(item => {
+                    return ({
+                        rowData: [
+                            <Checkbox
+                                ariaLabel='Select row'
+                                checked={checkedItems[item.productName] || false}
+                                name={item.productName}
+                                onChange={handleChange} />,
+                            <span>{item.productName}</span>,
+                            <FormInput ariaLabel='Product ID' defaultValue={item.productId}
```
Comment (prsdthkr) [inline src/Table/__stories__/Table.stories.js:231]:
```suggestion
                            <FormInput aria-label='Product ID' defaultValue={item.productId}
```

File: src/Table/__stories__/Table.stories.js [hunk 3]
```diff
@@ -120,3 +145,103 @@ export const richTable = () => {
...
+                                name={item.productName} />,
+                            <span>{item.quantity}</span>,
+                            <ObjectStatus status={item.status === 'Available' ? 'positive' : 'negative'}>{item.status}</ObjectStatus>,
+                            <Select options={suppliers} placeholder='Select a supplier'
+                                selectedKey={suppliers[suppliers.findIndex((supplier) => supplier.text === item.supplierName)].key}
+                                validationOverlayProps={{ 'aria-label': 'Supplier' }} />,
+                            <Link href={item.imageUrl}>Show image</Link>,
+                            <Checkbox ariaLabel='Heavy Weight' />,
+                            <FormInput ariaLabel='Categories' />,
```
Comment (prsdthkr) [inline src/Table/__stories__/Table.stories.js:240]:
```suggestion
                            <FormInput aria-label='Categories' />,
```

File: src/Table/__stories__/Table.stories.js [hunk 4]
```diff
@@ -120,3 +145,103 @@ export const richTable = () => {
...
+                                onChange={handleChange} />,
+                            <span>{item.productName}</span>,
+                            <FormInput ariaLabel='Product ID' defaultValue={item.productId}
+                                name={item.productName} />,
+                            <span>{item.quantity}</span>,
+                            <ObjectStatus status={item.status === 'Available' ? 'positive' : 'negative'}>{item.status}</ObjectStatus>,
+                            <Select options={suppliers} placeholder='Select a supplier'
+                                selectedKey={suppliers[suppliers.findIndex((supplier) => supplier.text === item.supplierName)].key}
+                                validationOverlayProps={{ 'aria-label': 'Supplier' }} />,
+                            <Link href={item.imageUrl}>Show image</Link>,
```
Comment (prsdthkr) [inline src/Table/__stories__/Table.stories.js:238]:
```suggestion
                            <Link aria-label={`Show image: ${item.productName}`}  href={item.imageUrl}>Show image</Link>,
```

**Review by prsdthkr — CHANGES_REQUESTED** [review]
Some a11y stuff:

File: src/Table/Table.js (hunk 1 above)
Comment (jacobdevera) [reply src/Table/Table.js:121]: I decided to change it to `__caption` in the end, but didn't put that change here yet . Thanks for catching that.

File: src/Table/Table.js [hunk 5]
```diff
@@ -44,17 +61,67 @@ const Table = React.forwardRef(({ headers, tableData, className, tableBodyClassN
...
     if (richTable) {
         checkboxHeader = <th className={tableCheckboxClasses}>{headers[0]}</th>;
-        displayHeaders = headers.splice(1, headers.length);
+        displayHeaders = displayHeaders.splice(1, headers.length);
     }
 
     return (
-        <table {...props} className={tableClasses}
-            ref={ref}>
+        <table {...props} aria-describedBy={captionId}
```
Comment (prsdthkr) [inline src/Table/Table.js:117]:
```suggestion
        <table {...props} aria-describedby={captionId}
```

File: src/Table/__stories__/Table.stories.js [hunk 6]
```diff
@@ -120,3 +145,103 @@ export const richTable = () => {
...
+                            <FormInput aria-label='Product ID' defaultValue={item.productId}
+                                name={item.productName} />,
+                            <span>{item.quantity}</span>,
+                            <ObjectStatus status={item.status === 'Available' ? 'positive' : 'negative'}>{item.status}</ObjectStatus>,
+                            <Select options={suppliers} placeholder='Select a supplier'
+                                selectedKey={suppliers[suppliers.findIndex((supplier) => supplier.text === item.supplierName)].key}
+                                validationOverlayProps={{ 'aria-label': 'Supplier' }} />,
+                            <Link aria-label={`Show image: ${item.productName}`} href={item.imageUrl}>Show image</Link>,
+                            <Checkbox ariaLabel='Heavy Weight' />,
+                            <FormInput aria-Label='Categories' />,
```
Comment (prsdthkr) [inline src/Table/__stories__/Table.stories.js:240]:
```suggestion
                            <FormInput aria-label='Categories' />,
```

File: src/Table/Table.js [hunk 7]
```diff
@@ -106,14 +177,29 @@ Table.displayName = 'Table';
...
+    /** Determines the type of keyboard navigation for the table. Set to `'cell'` for cell-level navigation or `'row'` for row-level navigation */
+    keyboardNavigation: PropTypes.oneOf(['none', 'cell', 'row']),
+    /** Localized text to be updated based on location/language */
+    localizedText: CustomPropTypes.i18n({
+        /** Localized string informing screen reader users the table can be navigated by arrow keys */
+        arrowKeys: PropTypes.string,
+        /** Localized string informing screen reader users the current cell can be edited  */
+        editModeEnable: PropTypes.string,
+        /** Localized string informing screen reader users how to return to cell navigation */
+        editModeDisable: PropTypes.string
```
Comment (prsdthkr) [inline src/Table/Table.js:201]:
```suggestion
        editModeDisable: PropTypes.string,
        row: PropTypes.string,
        column: PropTypes.string,
```

**Review by prsdthkr — CHANGES_REQUESTED** [review]
Some missing props and typos.

**Review by prsdthkr — APPROVED** [review]
LGTM 🥇

**Review by Mike-Diaz — APPROVED** [review]
LGTM. Keyboard nav works as expected!

**Pull request merged**